### PR TITLE
Add mnist example

### PR DESCRIPTION
**Pull request by apeforest**
- Add a mnist example
- Add the example to CI

### CODE REVIEW

File: examples/mxnet_mnist.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# Copyright 2017 Uber Technologies, Inc. All Rights Reserved.
```
Comment (yuxihu) [inline examples/mxnet_mnist.py]: 2017 => 2018 or 2019?

File: examples/mxnet_mnist.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+              num_epoch=args.epochs)  # train for at most 10 dataset passes
+
+# Step 5: Evaluate model accuracy
+test_iter = mx.io.NDArrayIter(mnist['test_data'], mnist['test_label'],
+                              batch_size)
+# predict accuracy of mlp
+acc = mx.metric.Accuracy()
+mlp_model.score(test_iter, acc)
+print(acc)
+assert acc.get()[1] > 0.94, "Achieved accuracy (%f) is lower than expected \
```
Comment (yuxihu) [inline examples/mxnet_mnist.py]: How do you decide the accuracy should be > 0.94? I just ran the example in a local container and it failed. The achieved acuracy is 0.9378. It might be flaky.

File: examples/mxnet_mnist.py (hunk 1 above)
Comment (apeforest) [reply examples/mxnet_mnist.py]: Removed this license since pytorch also did not have it.

File: examples/mxnet_mnist.py (hunk 2 above)
Comment (apeforest) [reply examples/mxnet_mnist.py]: The single node mnist example is `0.96`. I have changed the number of epochs to 10 and added initial param broadcasting. Hopefully we can always achieve this accuracy.
